ibis/backends/snowflake/registry.py

```diff
@@ -53,6 +53,8 @@ def _literal(t, op):
         return sa.func.object_construct_keep_null(
             *itertools.chain.from_iterable(value.items())
         )
+    elif dtype.is_uuid():
+        return sa.literal(str(value))
```

Comment:
Does this work? If so, we should apply it elsewhere in this PR where it makes sense.

```suggestion
        return sa.literal(str(value), type_=sa.UUID())
```

Reply:
Here we want to emulate a UUID type that is not supported by the database using the `STRING` type, so here it should be `type_=sa.Text()`, but SQLAlchemy is smart enough to know that.
https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html